From 19a68c112b37cb8ca8aa968a196c846e4410bce4 Mon Sep 17 00:00:00 2001
From: Hanna Hawa <hannah@marvell.com>
Date: Tue, 4 Jul 2017 22:40:33 +0300
Subject: [PATCH 1057/1345] arm64: dts: integrate a70x0 and a80x0 dts into
 mainline.

This commit refer to SDK commit 8d62991bb9e0("fix: dts: a70x0: a80x0:
align DTS name of  Armad-7/8k with mainline")

In SDK, armada-7040-db-A.dts and armada-8040-db-A.dts will be rename
as armada-7040-db.dts and armada-8040-db.dts. But in mainline kernel,
there has been the 2 dts file. So integrate SDK dts node to mainline
dts file.

Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/marvell/armada-7040-db.dts     |  190 ++++++++++++
 arch/arm64/boot/dts/marvell/armada-8040-db-A.dts   |    4 +-
 arch/arm64/boot/dts/marvell/armada-8040-db.dts     |  301 +++++++++++++++++++-
 .../boot/dts/marvell/armada-cp110-master.dtsi      |  118 ++++++++
 4 files changed, 604 insertions(+), 9 deletions(-)

diff --git a/arch/arm64/boot/dts/marvell/armada-7040-db.dts b/arch/arm64/boot/dts/marvell/armada-7040-db.dts
index 1244232..9b5aff2 100644
--- a/arch/arm64/boot/dts/marvell/armada-7040-db.dts
+++ b/arch/arm64/boot/dts/marvell/armada-7040-db.dts
@@ -59,6 +59,165 @@
 		device_type = "memory";
 		reg = <0x0 0x0 0x0 0x80000000>;
 	};
+
+	ap806 {
+		config-space@f0000000 {
+			serial@512000 {
+				status = "okay";
+			};
+			i2c@511000 {
+				status = "disabled";
+				clock-frequency = <100000>;
+			};
+			sd_mmc_reg0: regulator@0 {
+				gpios = <&gpio0 11 GPIO_ACTIVE_HIGH>;
+				gpios-states = <0>;
+				states = <1800000 0x1
+					3300000 0x0>;
+				enable-active-high;
+				status = "disabled";
+			};
+			sdhci@6e0000 {
+				status = "okay";
+				bus-width = <4>;
+				no-1-8-v;
+				non-removable; /* For now, need to configure the GPIO for card detect */
+			};
+		};
+	};
+
+	cp110-master {
+		config-space@f2000000 {
+			serial@702000 {
+				status = "disabled";
+			};
+			serial@702100 {
+				status = "disabled";
+			};
+			serial@702200 {
+				status = "disabled";
+			};
+			serial@702300 {
+				status = "disabled";
+			};
+			u3d@520000 {
+				status = "disabled";
+			};
+			udc@524100 {
+				status = "disabled";
+			};
+			mdio@12a200 {
+				#address-cells = <1>;
+				#size-cells = <0>;
+				phy2: ethernet-phy@2 {
+					reg = <0>;
+				};
+				phy3: ethernet-phy@3 {
+					reg = <1>;
+				};
+			};
+			gop {
+				emac0: mac0 {
+					phy-mode = "sfi"; /* lane-2 */
+					phys = <&comphy0 2 COMPHY_SFI>;
+					phy-names = "comphy";
+				};
+				emac2: mac2 {
+					phy-mode = "sgmii"; /* lane-0 */
+					phy = <&phy2>;
+					phys = <&comphy0 0 COMPHY_SGMII1>;
+					phy-names = "comphy";
+				};
+				emac3: mac3 {
+					phy-mode = "rgmii"; /* rgmii-1 */
+					phy = <&phy3>;
+				};
+			};
+			ppv22@000000 {
+				l4_chksum_jumbo_port = <0>;
+				eth0: eth0@010000 {
+					status = "okay";
+				};
+				eth1: eth1@020000 {
+					status = "okay";
+				};
+				eth2: eth2@030000 {
+					status = "okay";
+				};
+			};
+			comphy {
+				status = "okay";
+			};
+
+			nand@720000 {
+				nand-ecc-strength = <4>;
+				nand-ecc-step-size = <512>;
+				status = "disabled";
+				partition@0 {
+					label = "U-Boot";
+					reg = <0 0x200000>;
+				};
+				partition@200000 {
+					label = "Linux";
+					reg = <0x200000 0xd00000>;
+				};
+				partition@1000000 {
+					label = "Filesystem";
+					reg = <0x1000000 0x3f000000>;
+				};
+			};
+		};
+
+		pcie@0x640000 {
+			status = "okay";
+		};
+	};
+
+	usb3h0_phy: usb3-h0-phy {
+		compatible = "usb-nop-xceiv";
+		/* vcc-supply can be changed by io-expander or GPIO,
+		 * for A7040-DB we are using by default io-expander
+		 * for other board GPIO may be needed, should update
+		 * the phandle to gpio regulator
+		 */
+		vcc-supply = <&exp_usb3h0_vbus>;
+	};
+	usb3h1_phy: usb3-h1-phy {
+		compatible = "usb-nop-xceiv";
+		vcc-supply = <&exp_usb3h1_vbus>;
+	};
+	exp_usb3h0_vbus: usb3-0-vbus {
+		compatible = "regulator-fixed";
+		regulator-name = "usb3-0-vbus";
+		regulator-min-microvolt = <5000000>;
+		regulator-max-microvolt = <5000000>;
+		enable-active-high;
+		regulator-always-on;
+		gpio = <&expander0 0 GPIO_ACTIVE_HIGH>;
+	};
+	exp_usb3h1_vbus: usb3-1-vbus {
+		compatible = "regulator-fixed";
+		regulator-name = "usb3-1-vbus";
+		regulator-min-microvolt = <5000000>;
+		regulator-max-microvolt = <5000000>;
+		enable-active-high;
+		regulator-always-on;
+		gpio = <&expander0 1 GPIO_ACTIVE_HIGH>;
+	};
+	/* following GPIO entry is not used (for reference only)
+	 * GPIO is not connected in default A7040-DB board
+	 */
+	gpio_usb3_vbus: usb3-gpio-vbus {
+		compatible = "regulator-fixed";
+		pinctrl-names = "default";
+		pinctrl-0 = <&xhci0_vbus_pins>;
+		regulator-name = "xhci0-vbus";
+		regulator-min-microvolt = <5000000>;
+		regulator-max-microvolt = <5000000>;
+		enable-active-high;
+		gpio = <&cpm_gpio0 12 GPIO_ACTIVE_HIGH>;
+		status = "disabled";
+	};
 };
 
 &i2c0 {
@@ -105,6 +264,25 @@
 &cpm_i2c0 {
 	status = "okay";
 	clock-frequency = <100000>;
+	pinctrl-names = "default";
+	pinctrl-0 = <&i2c0_pins>;
+	eeprom@57 {
+		compatible = "at24,24c64";
+		reg = <0x57>;
+		pagesize = <32>;
+	};
+	eeprom@50 {
+		compatible = "at24,24c64";
+		reg = <0x50>;
+		pagesize = <32>;
+	};
+	expander0: pca9555@21 {
+		compatible = "nxp,pca9555";
+		pinctrl-names = "default";
+		gpio-controller;
+		#gpio-cells = <2>;
+		reg = <0x21>;
+	};
 };
 
 &cpm_spi1 {
@@ -137,14 +315,25 @@
 
 &cpm_sata0 {
 	status = "okay";
+
+	sata-port@0 {
+		status = "disabled";
+	};
+	sata-port@1 {
+		phys = <&comphy0 3 COMPHY_SATA1>;
+		phy-names = "comphy";
+		status = "okay";
+	};
 };
 
 &cpm_usb3_0 {
 	status = "okay";
+	usb-phy = <&usb3h0_phy>;
 };
 
 &cpm_usb3_1 {
 	status = "okay";
+	usb-phy = <&usb3h1_phy>;
 };
 
 &ap_sdhci0 {
@@ -159,6 +348,7 @@
 	bus-width = <4>;
 	no-1-8-v;
 	non-removable;
+	pinctrl-names = "default";
 };
 
 &cpm_mdio {
diff --git a/arch/arm64/boot/dts/marvell/armada-8040-db-A.dts b/arch/arm64/boot/dts/marvell/armada-8040-db-A.dts
index e6d1c84..0b5f09c 100644
--- a/arch/arm64/boot/dts/marvell/armada-8040-db-A.dts
+++ b/arch/arm64/boot/dts/marvell/armada-8040-db-A.dts
@@ -155,8 +155,8 @@
 			status = "okay";
 		};
 	};
-	cpn-110-slave {
-		config-space {
+	cp110-slave {
+		config-space@f4000000 {
 			/* usb3h0_1 is USB2 only (UTMI) */
 			usb3h0_1: usb3@500000 {
 				status = "okay";
diff --git a/arch/arm64/boot/dts/marvell/armada-8040-db.dts b/arch/arm64/boot/dts/marvell/armada-8040-db.dts
index dc0d084..d3c5534 100644
--- a/arch/arm64/boot/dts/marvell/armada-8040-db.dts
+++ b/arch/arm64/boot/dts/marvell/armada-8040-db.dts
@@ -59,15 +59,175 @@
 		device_type = "memory";
 		reg = <0x0 0x0 0x0 0x80000000>;
 	};
+
+	cp110-master {
+		config-space@f2000000 {
+			gop {
+				emac0: mac0 {
+					phy-mode = "sfi"; /* lane-2 */
+					phys = <&comphy0 2 COMPHY_SFI>;
+					phy-names = "comphy";
+				};
+				emac2: mac2 {
+					phy-mode = "none"; /* disabled */
+				};
+				emac3: mac3 {
+					phy-mode = "rgmii"; /* rgmii-1 */
+					phy = <&mdio0_phy1>;
+				};
+			};
+			ppv22@000000 {
+				l4_chksum_jumbo_port = <0>;
+				pinctrl-names = "default";
+				pinctrl-0 = <&cpm_ge2_rgmii_pins>;
+
+				eth0: eth0@010000 {
+					status = "okay";
+				};
+				eth1: eth1@020000 {
+					status = "disabled";
+				};
+				eth2: eth2@030000 {
+					status = "okay";
+				};
+			};
+
+			comphy {
+				status = "okay";
+			};
+
+			u3d@520000 {
+				/* There is no need to set pinctlr, since the
+				 * gpio is from io-expander
+				 */
+
+				/* To enable IO expander interrupts, SD/MMC on CP
+				 * Must be disabled in U-Boot & Linux
+				 * vbus-gpio = <&expander0 3 GPIO_ACTIVE_HIGH>;
+				 */
+			};
+		};
+		pcie@0x600000 {
+			status = "okay";
+		};
+		pcie@0x640000 {
+			status = "okay";
+		};
+
+		cpm_usb3h0_phy: cpm_usb3_phy0 {
+			compatible = "usb-nop-xceiv";
+			vcc-supply = <&cpm_reg_usb3h0_vbus>;
+		};
+
+		cpm_reg_usb3h0_vbus: cpm_usb3-vbus0 {
+			compatible = "regulator-fixed";
+			regulator-name = "cpm-usb3-vbus-h0";
+			regulator-min-microvolt = <5000000>;
+			regulator-max-microvolt = <5000000>;
+			enable-active-high;
+			gpio = <&expander0 0 GPIO_ACTIVE_HIGH>;
+		};
+
+		cpm_usb3h1_phy: cpm_usb3_phy1 {
+			compatible = "usb-nop-xceiv";
+			vcc-supply = <&cpm_reg_usb3h1_vbus>;
+		};
+
+		cpm_reg_usb3h1_vbus: cpm_usb3-vbus1 {
+			compatible = "regulator-fixed";
+			regulator-name = "cpm-usb3-vbus-h1";
+			regulator-min-microvolt = <5000000>;
+			regulator-max-microvolt = <5000000>;
+			enable-active-high;
+			gpio = <&expander0 1 GPIO_ACTIVE_HIGH>;
+		};
+	};
+
+	cp110-slave {
+		config-space@f4000000 {
+			mdio@12a200 {
+				#address-cells = <1>;
+				#size-cells = <0>;
+				mdio1_phy1: ethernet-phy@1 {
+					reg = <0>;
+				};
+			};
+			gop {
+				emac0_1: mac0 {
+					phy-mode = "sfi"; /* lane-2 */
+					phys = <&comphy1 2 COMPHY_SFI>;
+					phy-names = "comphy";
+				};
+				emac2_1: mac2 {
+					phy-mode = "rgmii"; /* rgmii-0 */
+					phy = <&mdio1_phy1>;
+				};
+				emac3_1: mac3 {
+					phy-mode = "none"; /* disabled */
+				};
+			};
+			ppv22@000000 {
+				l4_chksum_jumbo_port = <0>;
+				pinctrl-names = "default";
+				pinctrl-0 = <&cps_ge1_rgmii_pins>;
+
+				eth0_1: eth0@010000 {
+					status = "okay";
+				};
+				eth1_1: eth1@020000 {
+					status = "okay";
+				};
+				eth2_1: eth2@030000 {
+					status = "disabled";
+				};
+			};
+
+			comphy {
+				status = "okay";
+			};
+
+			nand@720000 {
+				nand-ecc-strength = <4>;
+				nand-ecc-step-size = <512>;
+				status = "disabled";
+				partition@0 {
+					label = "U-Boot";
+					reg = <0 0x200000>;
+				};
+				partition@200000 {
+					label = "Linux";
+					reg = <0x200000 0xd00000>;
+				};
+				partition@1000000 {
+					label = "Filesystem";
+					reg = <0x1000000 0x3f000000>;
+				};
+			};
+		};
+
+		cps_usb3h0_phy: cps_usb3_phy0 {
+			compatible = "usb-nop-xceiv";
+			vcc-supply = <&cps_reg_usb3h0_vbus>;
+		};
+
+		cps_reg_usb3h0_vbus: cps_usb3-vbus0 {
+			compatible = "regulator-fixed";
+			regulator-name = "cps-usb3-vbus-h0";
+			regulator-min-microvolt = <5000000>;
+			regulator-max-microvolt = <5000000>;
+			enable-active-high;
+			gpio = <&expander1 0 GPIO_ACTIVE_HIGH>;
+		};
+	};
 };
 
 &i2c0 {
-	status = "okay";
+	status = "disabled";
 	clock-frequency = <100000>;
 };
 
 &spi0 {
-	status = "okay";
+	status = "disabled";
 
 	spi-flash@0 {
 		#address-cells = <1>;
@@ -98,7 +258,6 @@
 	status = "okay";
 };
 
-
 /* CON5 on CP0 expansion */
 &cpm_pcie2 {
 	status = "okay";
@@ -107,21 +266,86 @@
 &cpm_i2c0 {
 	status = "okay";
 	clock-frequency = <100000>;
+	pinctrl-names = "default";
+	pinctrl-0 = <&cpm_i2c0_pins>;
+
+	/* U31 */
+	expander0: pca9555@21 {
+		compatible = "nxp,pca9555";
+		pinctrl-names = "default";
+		pinctrl-0 = <&pca0_pins>;
+		gpio-controller;
+		#gpio-cells = <2>;
+		reg = <0x21>;
+		/*
+		 * Support USB-DEVICE with SD in 8040
+		 * To enable IO expander interrupts, SD/MMC on CP
+		 * Must be disabled in U-Boot & Linux
+		 * interrupt-parent = <&cpm_gpio1>;
+		 * interrupts = <7 IRQ_TYPE_EDGE_FALLING>;
+		 * interrupt-controller;
+		 * #interrupt-cells = <2>;
+		 */
+	};
+
+	/* U25 */
+	expander1: pca9555@25 {
+		compatible = "nxp,pca9555";
+		pinctrl-names = "default";
+		gpio-controller;
+		#gpio-cells = <2>;
+		reg = <0x25>;
+		/*
+		 * Support USB-DEVICE with SD in 8040
+		 * To enable IO expander interrupts, SD/MMC on CP
+		 * Must be disabled in U-Boot & Linux
+		 * interrupt-parent = <&cpm_gpio1>;
+		 * interrupts = <7 IRQ_TYPE_EDGE_FALLING>;
+		 * gpio-controller;
+		 * #gpio-cells = <2>;
+		 * interrupt-controller;
+		 * #interrupt-cells = <2>;
+		 * reg = <0x25>;
+		 */
+	};
+
+	eeprom@57 {
+		compatible = "at24,24c64";
+		reg = <0x57>;
+		pagesize = <32>;
+	};
+	eeprom@50 {
+		compatible = "at24,24c64";
+		reg = <0x50>;
+		pagesize = <32>;
+	};
 };
 
 /* CON4 on CP0 expansion */
 &cpm_sata0 {
 	status = "okay";
+	sata-port@0 {
+		phys = <&comphy0 1 COMPHY_SATA0>;
+		phy-names = "comphy";
+		status = "okay";
+	};
+	sata-port@1 {
+		phys = <&comphy0 3 COMPHY_SATA1>;
+		phy-names = "comphy";
+		status = "okay";
+	};
 };
 
 /* CON9 on CP0 expansion */
 &cpm_usb3_0 {
 	status = "okay";
+	usb-phy = <&cpm_usb3h0_phy>;
 };
 
 /* CON10 on CP0 expansion */
 &cpm_usb3_1 {
 	status = "okay";
+	usb-phy = <&cpm_usb3h1_phy>;
 };
 
 &cpm_mdio {
@@ -150,33 +374,96 @@
 };
 
 &cps_i2c0 {
-	status = "okay";
+	status = "disabled";
 	clock-frequency = <100000>;
+
+	eeprom@57 {
+		compatible = "at24,24c64";
+		reg = <0x57>;
+		pagesize = <32>;
+	};
+	eeprom@50 {
+		compatible = "at24,24c64";
+		reg = <0x50>;
+		pagesize = <32>;
+	};
 };
 
 /* CON4 on CP1 expansion */
 &cps_sata0 {
 	status = "okay";
+	sata-port@0 {
+		phys = <&comphy1 1 COMPHY_SATA0>;
+		phy-names = "comphy";
+		status = "okay";
+	};
+	sata-port@1 {
+		phys = <&comphy1 3 COMPHY_SATA1>;
+		phy-names = "comphy";
+		status = "okay";
+	};
 };
 
 /* CON9 on CP1 expansion */
 &cps_usb3_0 {
 	status = "okay";
+	usb-phy = <&cps_usb3h0_phy>;
 };
 
 /* CON10 on CP1 expansion */
 &cps_usb3_1 {
-	status = "okay";
+	status = "disabled";
 };
 
 &ap_sdhci0 {
 	status = "okay";
-	bus-width = <4>;
+	bus-width = <4>;	/*SDK is <8>*/
 	non-removable;
+	marvell,xenon-emmc;
 };
 
 &cpm_sdhci0 {
 	status = "okay";
-	bus-width = <8>;
+	bus-width = <8>;	/*SDK is <4>*/
 	non-removable;
 };
+
+&cps_crypto {
+	status = "okay";
+};
+
+&cps_spi1 {
+	status = "okay";
+	pinctrl-names = "default";
+	pinctrl-0 = <&cps_spi1_pins>;
+
+	spi-flash@0 {
+		#address-cells = <0x1>;
+		#size-cells = <0x1>;
+		compatible = "jedec,spi-nor";
+		reg = <0x0>;
+		spi-max-frequency = <20000000>;
+		partition@0 {
+			label = "boot";
+			reg = <0x0 0x200000>;
+		};
+		partition@200000 {
+			label = "Filesystem";
+			reg = <0x200000 0xd00000>;
+		};
+		partition@d00000 {
+			label = "boot_2nd";
+			reg = <0xf00000 0x100000>;
+		};
+	};
+};
+
+&cps_pcie0 {
+	status = "okay";
+}
+&cps_pcie1 {
+	status = "okay";
+}
+&cps_pcie2 {
+	status = "okay";
+}
diff --git a/arch/arm64/boot/dts/marvell/armada-cp110-master.dtsi b/arch/arm64/boot/dts/marvell/armada-cp110-master.dtsi
index 603d286..c245a64 100644
--- a/arch/arm64/boot/dts/marvell/armada-cp110-master.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-cp110-master.dtsi
@@ -278,6 +278,124 @@
 					   <ICU_GRP_NSR 80 IRQ_TYPE_LEVEL_HIGH>,
 					   <ICU_GRP_NSR 79 IRQ_TYPE_LEVEL_HIGH>;
 			};
+
+			u3d@520000 {
+				compatible = "marvell,mvebu-u3d";
+				/* 0: 0x520000: USB 3.0 Device port 0: DEV_INFO_REG(0:15 - version_id)
+				 * 1: 0x121F10: Clock Source High reg: VBUS present indication in PHY SerDes lane#1
+				 */
+				reg = <0x520000 0x2000>, <0x121f10 0x40>;
+				interrupts = <ICU_GRP_NSR 102 IRQ_TYPE_LEVEL_HIGH>;
+				clocks = <&cpm_syscon0 1 24>;
+				dma-coherent;
+				status = "disabled";
+			};
+
+			udc@524100 {
+				compatible = "marvell,mv-udc";
+				reg = <0x524100 0x2000>;
+				interrupts = <ICU_GRP_NSR 102 IRQ_TYPE_LEVEL_HIGH>;
+				clocks = <&cpm_syscon0 1 24>;
+				dma-coherent;
+				status = "disabled";
+			};
+
+			ppv22@000000 {
+				compatible = "marvell,mv-pp22";
+				reg = <0x000000 0x90000>, /* Packet Processor regs */
+				      <0x129000 0x0600>, /* XMIB regs */
+				      <0x12a000 0x200>,  /* LED regs */
+				      <0x12a200 0x200>,  /* SMI regs */
+				      <0x12a400 0x200>,  /* TAI regs */
+				      <0x12a600 0x200>,  /* XSMI regs */
+				      <0x12b000 0x1000>,  /* MG Internal regs */
+				      <0x130000 0x6000>, /* MSPG regs */
+				      <0x130400 0x200>,  /* MSPG - XPCS regs */
+				      <0x130600 0x200>,  /* FCA - flow control regs*/
+				      <0x130e00 0x100>,  /* MSPG - GMAC regs */
+				      <0x130f00 0x100>,  /* MSPG - XLG MAC regs */
+				      <0x441100 0x100>;  /* RFU-1 Regs */
+				reg-names = "pp", "xmib", "led", "smi", "tai", "xsmi", "mg", "mspg", "xpcs",
+					    "fca", "gmac", "xlg", "rfu1";
+				clocks = <&cpm_syscon0 1 3>, <&cpm_syscon0 1 18>, <&cpm_syscon0 1 9>,
+					 <&cpm_syscon0 1 6>, <&cpm_syscon0 1 5>;
+				clock-names = "pp_clk", "gop_core_clk", "gop_clk", "mg_core_clk", "mg_clk";
+				status = "okay";
+				eth0: eth0@010000 {
+					interrupts = <ICU_GRP_NSR 39 IRQ_TYPE_LEVEL_HIGH>, /* TX CPU#0 Interrupt */
+						     <ICU_GRP_NSR 43 IRQ_TYPE_LEVEL_HIGH>, /* TX CPU#1 Interrupt */
+						     <ICU_GRP_NSR 47 IRQ_TYPE_LEVEL_HIGH>, /* TX CPU#2 Interrupt */
+						     <ICU_GRP_NSR 51 IRQ_TYPE_LEVEL_HIGH>, /* TX CPU#3 Interrupt */
+						     <ICU_GRP_NSR 55 IRQ_TYPE_LEVEL_HIGH>; /* RX Shared Interrupt */
+					port-id = <0>; /* pp2_port_id */
+					emac-data = <&emac0>;
+					status = "disabled";
+				};
+				eth1: eth1@020000 {
+					interrupts = <ICU_GRP_NSR 40 IRQ_TYPE_LEVEL_HIGH>, /* TX CPU#0 Interrupt */
+						     <ICU_GRP_NSR 44 IRQ_TYPE_LEVEL_HIGH>, /* TX CPU#1 Interrupt */
+						     <ICU_GRP_NSR 48 IRQ_TYPE_LEVEL_HIGH>, /* TX CPU#2 Interrupt */
+						     <ICU_GRP_NSR 52 IRQ_TYPE_LEVEL_HIGH>, /* TX CPU#3 Interrupt */
+						     <ICU_GRP_NSR 56 IRQ_TYPE_LEVEL_HIGH>; /* RX Shared Interrupt */
+					port-id = <1>; /* pp2_port_id */
+					emac-data = <&emac2>;
+					status = "disabled";
+				};
+				eth2: eth2@030000 {
+					interrupts = <ICU_GRP_NSR 41 IRQ_TYPE_LEVEL_HIGH>, /* TX CPU#0 Interrupt */
+						     <ICU_GRP_NSR 45 IRQ_TYPE_LEVEL_HIGH>, /* TX CPU#1 Interrupt */
+						     <ICU_GRP_NSR 49 IRQ_TYPE_LEVEL_HIGH>, /* TX CPU#2 Interrupt */
+						     <ICU_GRP_NSR 53 IRQ_TYPE_LEVEL_HIGH>, /* TX CPU#3 Interrupt */
+						     <ICU_GRP_NSR 57 IRQ_TYPE_LEVEL_HIGH>; /* RX Shared Interrupt */
+					port-id = <2>; /* pp2_port_id */
+					emac-data = <&emac3>;
+					status = "disabled";
+				};
+			};
+
+			gop {
+				emac0: mac0 {
+					interrupts = <ICU_GRP_NSR 129 IRQ_TYPE_LEVEL_HIGH>;  /* Link IRQ */
+					mac-address = [00 00 00 00 00 01];
+					port-id = <0>; /* gop_port_id */
+				};
+				emac2: mac2 {
+					interrupts = <ICU_GRP_NSR 128 IRQ_TYPE_LEVEL_HIGH>;  /* Link IRQ */
+					mac-address = [00 00 00 00 00 02];
+					port-id = <2>; /* gop_port_id */
+				};
+				emac3: mac3 {
+					interrupts = <ICU_GRP_NSR 127 IRQ_TYPE_LEVEL_HIGH>;  /* Link IRQ */
+					mac-address = [00 00 00 00 00 03];
+					port-id = <3>; /* gop_port_id */
+				};
+			};
+
+			comphy0: comphy {
+				compatible = "marvell,cp110-comphy";
+				reg = <0x441000 0x100>, <0x120000 0x6000>;
+				reg-names = "comphy", "serdes";
+				dfx-reg = <0x400280 4>;
+				#phy-cells = <2>;
+				sar-data = <&cpm_sar>;
+				skip_pcie_power_off;
+				status = "disabled";
+			};
+
+			nand@720000 {
+				compatible = "marvell,armada370-nand";
+				reg = <0x720000 0x54>;
+				#address-cells = <1>;
+				#size-cells = <1>;
+				num-cs = <1>;
+				marvell,nand-enable-arbiter;
+				nand-on-flash-bbt;
+				interrupts = <ICU_GRP_NSR 115 IRQ_TYPE_LEVEL_HIGH>;
+				clocks = <&cpm_syscon0 1 2>;
+				nand-ecc-strength = <4>;
+				nand-ecc-step-size = <512>;
+				status = "disabled";
+			};
 		};
 
 		cpm_pcie0: pcie@f2600000 {
-- 
1.7.9.5

